File: MatchNBuy.Model/Message.cs
using System;
using System.ComponentModel.DataAnnotations;
using essentialMix.Data.Model;

namespace MatchNBuy.Model
{
	[Serializable]
    public class Message : IEntity
	{
		[Key]
		public Guid Id { get; set; }

		[StringLength(128, MinimumLength = 128)]
		public string ThreadId { get; set; }

		public virtual Thread Thread { get; set; }

		[Required]
		[StringLength(128, MinimumLength = 128)]
		public string SenderId { get; set; }

		public virtual User Sender { get; set; }

		[Required]
		[StringLength(128, MinimumLength = 128)]
		public string RecipientId { get; set; }

		public virtual User Recipient { get; set; }

		[Required]
        [StringLength(128)]
        public string Subject { get; set; }

		[Required]
        [StringLength(512)]
        public string Content { get; set; }

        public DateTime? DateRead { get; set; }
        public DateTime MessageSent { get; set; }
        public bool SenderDeleted { get; set; }
        public bool RecipientDeleted { get; set; }
        public bool IsArchived { get; set; }
    }
}

File: DatingApp-SPA/src/app/data/model/Message.ts
import { IUserForLogin } from "./User";

export interface IMessageThread {
	threadId: string;
	participant: IUserForLogin;
	isRead: boolean;
	lastModified: Date;
	count: number;
}

export interface IMessage {
	id: string;
	threadId: string;
	senderId: string;
	recipientId: string;
	subject: string;
	content: string;
	isRead: boolean;
	dateRead?: Date;
	messageSent: Date;
	senderDeleted: boolean;
	recipientDeleted: boolean;
	isArchived: boolean;
}

export interface IMessageToAdd {
	recipientId: string;
	content: string;
}

export interface IMessageToEdit {
	content: string;
}
